File: t/CS/AnimationDemo/UI/ViewerForm.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools;
using Leadtools.Demos;
using Leadtools.Controls;

using Leadtools.Drawing;

namespace AnimationDemo
{
   public partial class ViewerForm : Form
   {
      public ViewerForm( )
      {
         InitializeComponent();

         InitClass();
      }

      private RasterPictureBox _viewer;
      private string _name;

      private void InitClass( )
      {
         _viewer = new RasterPictureBox();
         _viewer.Dock = DockStyle.Fill;
         _viewer.BorderStyle = BorderStyle.None;
         _viewer.FrameChanged += new EventHandler<RasterPictureBoxFrameChangedEventArgs>(_viewer_FrameChanged);
         Controls.Add(_viewer);
         _viewer.BringToFront();
      }

      void _viewer_FrameChanged(object sender, RasterPictureBoxFrameChangedEventArgs e)
      {
         UpdateCaption();
         if (MdiParent != null)
            ((MainForm)MdiParent).UpdateControls();
       }

      public void Initialize(ImageInformation info, RasterPaintProperties paintProperties, bool animateRegions, bool snap, bool useDpi, RasterPictureBoxAnimationMode animationMode)
      {
         _viewer.BeginUpdate();
         UpdatePaintProperties(paintProperties);
         _viewer.Image = info.Image;
         _viewer.UseDpi = useDpi;
         _viewer.AnimationMode = animationMode;
         if(_viewer.Image != null)
            _viewer.Image.Changed += new EventHandler<RasterImageChangedEventArgs>(Image_Changed);

         if (Image.Width > this.Viewer.Size.Width || Image.Height > this.Viewer.Size.Height)
            this.Viewer.SizeMode = RasterPictureBoxSizeMode.Fit;

         _name = info.Name;
         UpdateCaption();
         _viewer.EndUpdate();
      }

      public void UpdatePaintProperties(RasterPaintProperties paintProperties)
      {
         _viewer.PaintProperties = paintProperties;
         _viewer.Refresh();
      }

      private void UpdateCaption( )
      {
         Text = string.Format("{0} - Page {1}:{2}", _name, _viewer.Image.Page, _viewer.Image.PageCount);
      }

      public RasterImage Image
      {
         get
         {
            return _viewer.Image;
         }
      }

      public RasterPictureBox Viewer
      {
         get
         {
            return _viewer;
         }
      }

      public string Title
      {
         get
         {
            return _name;
         }
      }

      private void Image_Changed(object sender, RasterImageChangedEventArgs e)
      {
         UpdateCaption();
         if(MdiParent != null)
            ((MainForm)MdiParent).UpdateControls();
      }
   }
}

File: t/VB/AnimationDemo/UI/ViewerForm.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************

Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools
Imports Leadtools.Demos
Imports Leadtools.Controls

Imports Leadtools.Drawing

Namespace AnimationDemo
   Partial Public Class ViewerForm
      Inherits Form
      Public Sub New()
         InitializeComponent()

         InitClass()
      End Sub

      Private _viewer As RasterPictureBox
      Private _name As String

      Private Sub InitClass()
         _viewer = New RasterPictureBox()
         _viewer.Dock = DockStyle.Fill
         _viewer.BorderStyle = BorderStyle.None
         AddHandler _viewer.FrameChanged, AddressOf _viewer_FrameChanged
         Controls.Add(_viewer)
         _viewer.BringToFront()
      End Sub

      Private Sub _viewer_FrameChanged(sender As Object, e As RasterPictureBoxFrameChangedEventArgs)
         UpdateCaption()
         If MdiParent IsNot Nothing Then
            CType(MdiParent, MainForm).UpdateControls()
         End If
      End Sub

      Public Sub Initialize(info As ImageInformation, paintProperties As RasterPaintProperties, animateRegions As Boolean, snap As Boolean, useDpi As Boolean, animationMode As RasterPictureBoxAnimationMode)
         _viewer.BeginUpdate()
         UpdatePaintProperties(paintProperties)
         _viewer.Image = info.Image
         _viewer.UseDpi = useDpi
         _viewer.AnimationMode = animationMode
         If _viewer.Image IsNot Nothing Then
            AddHandler _viewer.Image.Changed, AddressOf Image_Changed
         End If

         If Image.Width > Me.Viewer.Size.Width OrElse Image.Height > Me.Viewer.Size.Height Then
            Me.Viewer.SizeMode = RasterPictureBoxSizeMode.Fit
         End If

         _name = info.Name
         UpdateCaption()
         _viewer.EndUpdate()
      End Sub

      Public Sub UpdatePaintProperties(paintProperties As RasterPaintProperties)
         _viewer.PaintProperties = paintProperties
         _viewer.Refresh()
      End Sub

      Private Sub UpdateCaption()
         Text = String.Format("{0} - Page {1}:{2}", _name, _viewer.Image.Page, _viewer.Image.PageCount)
      End Sub

      Public ReadOnly Property Image() As RasterImage
         Get
            Return _viewer.Image
         End Get
      End Property

      Public ReadOnly Property Viewer() As RasterPictureBox
         Get
            Return _viewer
         End Get
      End Property

      Public ReadOnly Property Title() As String
         Get
            Return _name
         End Get
      End Property

      Private Sub Image_Changed(sender As Object, e As RasterImageChangedEventArgs)
         UpdateCaption()
         If MdiParent IsNot Nothing Then
            CType(MdiParent, MainForm).UpdateControls()
         End If
      End Sub
   End Class
End Namespace
